tests/utils/stl/test/format.py

```diff
@@ -10,312 +10,310 @@
+        shared.env['TMPDIR'] = execDir
+        shared.env['TEMPDIR'] = execDir
+
+        return [
+            ('Build setup', self.getBuildSetupSteps(test, litConfig, shared)),
+            ('Build', self.getBuildSteps(test, litConfig, shared)),
+            ('Intellisense response file', self.getIsenseRspFileSteps(test, litConfig, shared)),
+            ('Test setup', self.getTestSetupSteps(test, litConfig, shared)),
+            ('Test', self.getTestSteps(test, litConfig, shared))]
+
+    def getBuildSetupSteps(self, test, litConfig, shared):
+        shutil.rmtree(shared.execDir, ignore_errors=True)
+        Path(shared.execDir).mkdir(parents=True, exist_ok=True)
+
+        # Makes this function a generator which yields nothing
+        yield from []
+
+    def getBuildSteps(self, test, litConfig, shared):
+        filename = test.path_in_suite[-1]
+        _, tmpBase = test.getTempPaths()
+
+        shouldFail = TestType.FAIL in test.testType
+        if TestType.COMPILE in test.testType:
+            cmd = [test.cxx, '/c', test.getSourcePath(), *test.flags, *test.compileFlags]
+            yield TestStep(cmd, shared.execDir, shared.env, shouldFail)
+        elif TestType.LINK in test.testType:
+            objFile = tmpBase + '.o'
+            cmd = [test.cxx, '/c', test.getSourcePath(), *test.flags, *test.compileFlags, '/Fo' + objFile]
+            yield TestStep(cmd, shared.execDir, shared.env, False)
+
+            exeFile = tmpBase + '.exe'
+            cmd = [test.cxx, objFile, *test.flags, '/Fe' + exeFile, '/link', *test.linkFlags]
+            yield TestStep(cmd, shared.execDir, shared.env, shouldFail)
+        elif TestType.RUN in test.testType:
+            shared.execFile = tmpBase + '.exe'
+            if 'kernel' in test.requires:
+                name = str(shared.execFile).replace('\\','.').replace(':','.')
+
+                test.compileFlags.append([
+                    '/DKERNEL_TEST_NAME=L"' + name + '"',
+                    '/FIstl_kernel/kernel_test_constants.h',
+                    '/I' + litConfig.utils_dir + '/kernel/inc',
+                    '/I' + litConfig.wdk_include + '/km',
+                    #'/I' + litConfig.wdk_include + '/km/crt', #causes vadefs.h conflicts
+                    '/I' + litConfig.wdk_include + '/shared',
+                ])
+                test.linkFlags.append([
+                    '/LIBPATH:' + litConfig.wdk_lib + '/km/' + litConfig.target_arch,
+                    '/IGNORE:4210',
+                    '/machine:'+litConfig.target_arch,
+                    '/entry:DriverEntry',
+                    '/subsystem:native',
+                    '/nodefaultlib',
+                    'stl_kernel.lib',
+                    'BufferOverflowFastFailK.lib',
+                    'ntoskrnl.lib',
+                    'hal.lib',
+                    'wmilib.lib',
+                    'Ntstrsafe.lib',
+                    'libcpmt.lib',
+                    'libcmt.lib',
+                ])

```

Comment:
note that all of the flags except for /machine will migrate to an envlst file as soon as we make one.